src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -387,32 +388,70 @@ internal static DataViewType LoadType(ModelLoadContext ctx)
                 Contracts.AssertValue(ctx);
                 // *** Binary format ***
                 //   - bool: is vector
-                //   - int: vector size
-                //   - byte: ItemKind of input column (only R4 and R8 are valid)
                 bool isVector = ctx.Reader.ReadBoolean();
-                int vectorSize = ctx.Reader.ReadInt32();
-                Contracts.CheckDecode(vectorSize >= 0);
-                Contracts.CheckDecode(vectorSize > 0 || !isVector);
+
+                if (ctx.Header.ModelVerWritten < 0x00010002)
+                {
+                    // *** Previous Binary format ***
```

Comment:
Generally, it is good to add a test case for the "previous" format, so we know it still loads correctly.

In the past, we've added saved models here: https://github.com/dotnet/machinelearning-testdata/tree/master/Microsoft.ML.TestModels/TestModels

Reply:
Good idea. Thanks for the feedback.

There's [another test](https://github.com/dotnet/machinelearning/blob/d3dca2e952f25e5d445dda0383d2b9765bf6625b/test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs#L778) that checks for backward compatibility with an even older version of the NormalizerTransformer. I noticed that in that test they actually have the model with the 'previous older' format inside this other folder: https://github.com/dotnet/machinelearning/tree/master/test/data/backcompat

So maybe its better to use that folder instead, for this specific case?

Reply:
Checking in binary files into our code repo becomes a problem by bloating our git history. That’s why we made the testdata repo.